packages/zowe-explorer/src/extension.ts

```diff
@@ -66,16 +66,22 @@ export async function activate(context: vscode.ExtensionContext): Promise<ZoweEx
     // Determine the runtime framework to support special behavior for Theia
     globals.defineGlobals(preferencesTempPath);
 
-    // Call cleanTempDir before continuing
-    // this is to handle if the application crashed on a previous execution and
-    // VSC didn't get a chance to call our deactivate to cleanup.
-    await deactivate();
-
     try {
-        fs.mkdirSync(globals.ZOWETEMPFOLDER);
-        fs.mkdirSync(globals.ZOWE_TMP_FOLDER);
-        fs.mkdirSync(globals.USS_DIR);
-        fs.mkdirSync(globals.DS_DIR);
+        if (!fs.existsSync(globals.ZOWETEMPFOLDER)) {
```

Comment:
Just out of curiosity: why we are using the synchronous API to work with the file system?

Just a suggestion: Maybe we can use the [async API](https://code.visualstudio.com/api/references/vscode-api#FileSystem), which is built in VSCode?

The benefits will be: better user experience, better resources consumption and of course, this VSCode API will work fine in Theia, in Eclipse Che or in some browser VSCode versions with ZE installed. It will also handle potential problems between the operating systems, because it is basically another layer of abstraction above the usual file system API.